janusgraph-server/src/test/java/org/janusgraph/graphdb/server/JanusGraphServerTest.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.server;
+
+import org.apache.tinkerpop.gremlin.server.GremlinServer;
+import org.apache.tinkerpop.gremlin.server.Settings;
+import org.janusgraph.graphdb.database.StandardJanusGraph;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class JanusGraphServerTest {
+    @Test
+    public void testGremlinServerIsCorrectlyLoadedWithExpectedGraphs() {
+        final JanusGraphServer server = new JanusGraphServer("src/test/resources/janusgraph-server-with-serializers.yaml");
+
+        CompletableFuture<Void> start = server.start();
+
+        assertFalse(start.isCompletedExceptionally());
+
+        GremlinServer gremlinServer = server.getGremlinServer();
+        StandardJanusGraph graph = (StandardJanusGraph) gremlinServer.getServerGremlinExecutor().getGraphManager().getGraph("graph");
+        assertNotNull(graph);
+
+        CompletableFuture<Void> stop = server.stop();
+        CompletableFuture.allOf(start, stop).join();
+    }
+
+    @Test
+    public void testSerializersAreConfigured() {
+        final JanusGraphServer server = new JanusGraphServer("src/test/resources/janusgraph-server-without-serializers.yaml");
+
+        CompletableFuture<Void> start = server.start();
+
+        assertFalse(start.isCompletedExceptionally());
+
+        Settings settings = server.getGremlinSettings();
+        Optional<Settings.SerializerSettings> graphBinary = settings.serializers
+            .stream()
+            .filter(it -> it.className.equals("org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0"))
+            .skip(1).findFirst();
+
+        assertTrue(graphBinary.isPresent());
```

Comment:
What are you trying to do here? Skip one entry in the stream and check if there's a second one? You can also use `count()` on the stream then.

Reply:
It should ensure the automatic configuration works. I'll refactor the solution a bit.